extensions/blocks/contact-form/components/jetpack-field.js

```diff
@@ -3,60 +3,117 @@
+	const {
+		isSelected,
+		type,
+		required,
+		label,
+		parentBlock,
+		setAttributes,
+		placeholder,
+		padding,
+		spacing,
+	} = props;
+
+	useEffect( () => {
+		if ( parentBlock && parentBlock.attributes.padding !== padding ) {
+			setAttributes( { padding: parentBlock.attributes.padding } );
+		}
+
+		if ( parentBlock && parentBlock.attributes.spacing !== spacing ) {
+			setAttributes( { spacing: parentBlock.attributes.spacing } );
+		}
+	} );
 
-function JetpackField( {
-	isSelected,
-	type,
-	required,
-	label,
-	setAttributes,
-	defaultValue,
-	placeholder,
-	id,
-} ) {
 	return (
 		<Fragment>
 			<div className={ classNames( 'jetpack-field', { 'is-selected': isSelected } ) }>
-				<TextControl
-					type={ type }
-					label={
-						<JetpackFieldLabel
-							required={ required }
-							label={ label }
-							setAttributes={ setAttributes }
-							isSelected={ isSelected }
-						/>
-					}
-					placeholder={ placeholder }
-					value={ placeholder }
-					onChange={ value => setAttributes( { placeholder: value } ) }
-					title={ __( 'Set the placeholder text', 'jetpack' ) }
+				<JetpackFieldLabel
+					required={ required }
+					label={ label }
+					setAttributes={ setAttributes }
+					isSelected={ isSelected }
 				/>
+				<Disabled>
+					<TextControl
+						type={ type }
+						placeholder={ placeholder }
+						value={ placeholder }
+						onChange={ value => setAttributes( { placeholder: value } ) }
+						title={ __( 'Set the placeholder text', 'jetpack' ) }
+						style={ {
+							padding: padding + 'px',
+							marginBottom: spacing + 'px',
+						} }
+					/>
+				</Disabled>
 			</div>
+
+			<BlockControls>
+				<Toolbar>
+					<ToolbarButton
+						title={ __( 'Required' ) }
+						icon={ renderMaterialIcon(
+							<Path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z" />
+						) }
+						onClick={ () => {
+							setAttributes( { required: ! required } );
+						} }
+						className={ required ? 'is-pressed' : undefined }
+					/>
+				</Toolbar>
+			</BlockControls>
+
 			<InspectorControls>
 				<PanelBody title={ __( 'Field Settings', 'jetpack' ) }>
-					<TextControl
-						label={ __( 'Default Value', 'jetpack' ) }
-						value={ defaultValue }
-						onChange={ value => setAttributes( { defaultValue: value } ) }
-					/>
-					<TextControl
-						label={ __( 'ID', 'jetpack' ) }
-						value={ id }
-						onChange={ value => setAttributes( { id: value } ) }
+					<ToggleControl
+						label={ __( 'Field is required', 'jetpack' ) }
+						className="jetpack-field-label__required"
+						checked={ required }
+						onChange={ value => setAttributes( { required: value } ) }
+						help={ __(
+							'Does this field have to be completed for the form to be submitted?',
+							'jetpack'
+						) }
 					/>
 				</PanelBody>
 			</InspectorControls>
```

Comment:
What about moving the `BlockControls` and `InspectorControls` for the required setting in an external utility and import it everywhere instead of repeating the whole thing?

Reply:
Much better, created a `JetpackFieldControls` component that removes this duplication.